osu.Game/Overlays/Comments/VotePill.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Allocation;
+using osu.Game.Graphics.Sprites;
+using osuTK.Graphics;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using System.Collections.Generic;
+using osuTK;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Framework.Bindables;
+using System.Linq;
+
+namespace osu.Game.Overlays.Comments
+{
+    public class VotePill : LoadingButton, IHasAccentColour
+    {
+        public Color4 AccentColour { get; set; }
+
+        protected override IEnumerable<Drawable> EffectTargets => null;
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        private readonly Comment comment;
+        private Box background;
+        private Box hoverLayer;
+        private CircularContainer borderContainer;
+        private SpriteText sideNumber;
+        private OsuSpriteText votesCounter;
+        private CommentVoteRequest request;
+
+        private readonly BindableBool isVoted = new BindableBool();
+
+        public VotePill(Comment comment)
+        {
+            this.comment = comment;
+            setCount(comment.VotesCount);
+
+            Action = onAction;
+
+            AutoSizeAxes = Axes.X;
+            Height = 20;
+            LoadingAnimationSize = new Vector2(10);
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            AccentColour = borderContainer.BorderColour = sideNumber.Colour = colours.GreenLight;
+            hoverLayer.Colour = Color4.Black.Opacity(0.5f);
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            isVoted.Value = comment.IsVoted;
+            isVoted.BindValueChanged(voted => background.Colour = voted.NewValue ? AccentColour : OsuColour.Gray(0.05f), true);
+        }
+
+        private void onAction()
+        {
+            request = new CommentVoteRequest(comment.Id, isVoted.Value ? CommentVoteAction.UnVote : CommentVoteAction.Vote);
+            request.Success += onSuccess;
+            api.Queue(request);
+        }
+
+        private void onSuccess(CommentBundle response)
+        {
+            isVoted.Value = !isVoted.Value;
+            setCount(response.Comments.First().VotesCount);
```

Comment:
How about doing `.Single()` instead? Is there any threat this will ever return more than one comment in the response?